docs/USAGE.md

```diff
@@ -54,8 +58,56 @@ In order to build test plans, Testground needs to know where its source director
 
 For special cases, supply the `TESTGROUND_SRCDIR` environment variable.
 
+## Starting a testground daemon
+
+---
+
+⚠️ In the past, if the `[client]` section in the `.env.toml` file was not set
+up, we would start an embedded daemon to service the request. To avoid
+confusion, we no longer do that.
+
+---
+
+Testground has a daemon/client architecture:
+
+* The daemon performs the heavy-lifting. It populates a catalogue of test plans,
+  performs builds, and schedules runs, amongst other things.
+    * The daemon is intended to run in a server setting, but can of course be
+      run locally in a fashion similar to the IPFS daemon/client CLI. 
+    * The daemon exposes an HTTP API to receive client commands.
+* The client is a lightweight CLI tool that sends commands to the daemon via its
+  HTTP API.
```

Comment:
This section is duplicating the `OVERVIEW.md` - if we want to merge this architecture part here, we should remove `OVERVIEW.md`.

Reply:
I think we should just remove it from here, and add architectural docs under other doc, and not under USAGE.md - USAGE.md is our `Getting Started` guide, and should be minimal I think.

Reply:
Over-documenting is a nice problem to have. Let's go that path now, and dry down/consolidate/refine/polish the docs later.

Reply:
OK, fair enough.